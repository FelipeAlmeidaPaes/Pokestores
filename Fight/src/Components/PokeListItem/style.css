li.poke {
  position: relative;
  width: auto;
  height: fit-content;
  background: #fff;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 20px 10px;
  min-width: 168px;
  transition: .5s ease;
  border: 1px solid #f3f3f3
}

li.poke:hover {
  background: #f3f3f3;
  box-shadow: 0px 0px 5px #505050
}

span.tooltip-text {
  bottom: -30px;
  position: absolute;
  left: 0;
  width: 100%;
  height: -moz-fit-content;
  height: fit-content;
  background: #fff;
  transition: .3s;
  z-index: 2;
  padding: 5px;
  box-shadow: 0 0 5px black;
  font-size: 12px;
  display: none;
  box-sizing: border-box;
}

span.tooltip-text:before {
  content: '';
  height: 8px;
  width: 8px;
  background: #fff;
  position: absolute;
  top: -4px;
  left: 50%;
  transform: rotate(45deg);
}

.poke:hover .tooltip-text {
  display: block;
}

.image {
  max-width: 96px;
}

.name:first-letter {
  text-transform: uppercase;
}

.price {
  color: #be0000;
}

.addCartButton {
  width: 100%;
  height: 40px;
  background: linear-gradient(to bottom, #be0000 0%, #252525 100%);
  border: none;
  border-radius: 5px;
  position: relative;
  color: #fbfbfb;
  font-weight: 600;
  font-family: 'Open Sans', sans-serif;
  text-shadow: 1px 1px 1px rgba(0, 0, 0, .4);
  font-size: 15px;
  text-align: left;
  text-indent: 5px;
  box-shadow: 0px 3px 0px 0px rgba(0, 0, 0, .2);
  cursor: pointer;
  margin-top: 5px;
}

.addCartButton:after {
  content: "";
  width: 0;
  height: 0;
  display: block;
  border-top: 20px solid #252525;
  border-bottom: 20px solid #252525;
  border-left: 16px solid transparent;
  border-right: 20px solid #252525;
  position: absolute;
  opacity: 0.6;
  right: 0;
  top: 0;
  border-radius: 0 5px 5px 0;
}

p,
label {
  color: #5a5555;
  margin: 0;
}