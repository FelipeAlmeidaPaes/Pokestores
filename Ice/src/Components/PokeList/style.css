  #searchInput {
    position: absolute;
    top: 0;
    left: 0;
    height: 40px;
    width: 100%;
    box-sizing: border-box;
    padding: 0 40px;
    border-right: none;
    border-left: none;
    border-top: none;
    outline: none;
    background-image: url('../../Assets/search-icon.png');
    background-repeat: no-repeat;
    background-position: left center;
    background-size: contain;
    opacity: .7;
    transition: .3s linear;
  }
  
  #searchInput:focus {
    opacity: 1;
    transition: .3s linear;
  }
  
  ul#pokelist {
    display: grid;
    grid-template-areas: 'a a a a a a';
    grid-auto-rows: 208px;
    grid-gap: 20px;
    margin: 60px auto;
    padding: 0;
    height: calc(100vh - 80px);
    overflow-y: scroll;
    padding: 2px;
  }
  
  @media only screen and (min-width: 320px) and (max-width: 480px) {
    ul#pokelist {
      grid-template-areas: 'a';
      margin-bottom: 0;
      max-height: 440px;
    }
  }
  
  @media only screen and (min-width: 481px) and (max-width: 600px) {
    ul#pokelist {
      grid-template-areas: 'a a';
      margin-bottom: 0;
      max-height: 440px;
    }
  }
  
  @media only screen and (min-width: 601px) and (max-width: 768px) {
    ul#pokelist {
      grid-template-areas: 'a a a';
      margin-bottom: 0;
      max-height: 440px;
    }
  }
  
  @media only screen and (min-width: 768px) and (max-width: 900px) {
    ul#pokelist {
      grid-template-areas: 'a a a a';
      margin-bottom: 0;
      max-height: 440px;
    }
  }
  
  @media only screen and (min-width: 901px) and (max-width: 1024px) {
    ul#pokelist {
      grid-template-areas: 'a a a a';
      margin-bottom: 0;
      max-height: 440px;
    }
  }
  
  @media only screen and (min-width: 1025px) and (max-width: 1200px) {
    ul#pokelist {
      grid-template-areas: 'a a a';
      margin-bottom: 0;
      max-height: calc(100vh - 80px)
    }
  }
  
  @media only screen and (min-width: 1201px) and (max-width: 1800px) {
    ul#pokelist {
      grid-template-areas: 'a a a a';
      margin-bottom: 0;
      max-height: calc(100vh - 80px)
    }
  }