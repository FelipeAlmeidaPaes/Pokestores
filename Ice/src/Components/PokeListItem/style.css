li.poke {
  position: relative;
  width: auto;
  height: fit-content;
  background: #f3f3f3;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 20px 10px;
  min-width: 168px;
  box-shadow: 0px 0px 5px #505050
}

span.tooltip-text {
  bottom: -30px;
  position: absolute;
  left: 0;
  width: 100%;
  height: -moz-fit-content;
  height: fit-content;
  background: #fff;
  transition: .3s;
  z-index: 2;
  padding: 5px;
  box-shadow: 0 0 5px black;
  font-size: 12px;
  display: none;
  box-sizing: border-box;
}

span.tooltip-text:before {
  content: '';
  height: 8px;
  width: 8px;
  background: #fff;
  position: absolute;
  top: -4px;
  left: 50%;
  transform: rotate(45deg);
}

.poke:hover .tooltip-text {
  display: block;
}

.image {
  max-width: 96px;
}

.name:first-letter {
  text-transform: uppercase;
}

.addCartButton {
  width: 100%;
  height: 40px;
  background: linear-gradient(to bottom, #4eb5e5 0%, #389ed5 100%);
  border: none;
  border-radius: 5px;
  position: relative;
  border-bottom: 4px solid #2b8bc6;
  color: #fbfbfb;
  font-weight: 600;
  font-family: 'Open Sans', sans-serif;
  text-shadow: 1px 1px 1px rgba(0, 0, 0, .4);
  font-size: 15px;
  text-align: left;
  text-indent: 5px;
  box-shadow: 0px 3px 0px 0px rgba(0, 0, 0, .2);
  cursor: pointer;
  margin-top: 5px;
}

.addCartButton:after {
  content: "";
  width: 0;
  height: 0;
  display: block;
  border-top: 20px solid #187dbc;
  border-bottom: 20px solid #187dbc;
  border-left: 16px solid transparent;
  border-right: 20px solid #187dbc;
  position: absolute;
  opacity: 0.6;
  right: 0;
  top: 0;
  border-radius: 0 5px 5px 0;
}

p {
  margin: 0;
}