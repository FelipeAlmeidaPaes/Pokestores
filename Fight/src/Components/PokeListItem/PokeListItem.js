import React, { useState } from "react";
import _ from "lodash";
import './style.css';

function PokeListItem({ pokemon, addCart }) {
  const [quantity, setQuantity] = useState(1);
  const max_quantity = 6;
  const name = pokemon.name;
  const image = pokemon.sprites["front_default"];
  const price = pokemon.order + ",00";
  const abilities = pokemon.abilities.map((ability) => {
    return ability.ability.name;
  });

  return (
    <>
      <li className="poke">
        <img className="image" src={image ? image : ""} alt={"pokemon.img"} />
        <p className="name">{name}</p>
        <p className="price">
          <b>R$ {price}</b>
        </p>
        <div className="form-control">
          <label>qtd:</label>
          <select
            className="select__quantity"
            onChange={(e) => setQuantity(e.target.value)}
          >
            {_.range(1, max_quantity + 1).map((value) => (
              <option key={value} value={value}>
                {value}
              </option>
            ))}
          </select>
        </div>
        <button
          className="addCartButton"
          onClick={() => addCart(name, price, image, quantity)}
        >
          Adicionar
        </button>

        <span className="tooltip-text">
          habilidades: {abilities.join(", ")}
        </span>
      </li>
    </>
  );
}

export default PokeListItem;
