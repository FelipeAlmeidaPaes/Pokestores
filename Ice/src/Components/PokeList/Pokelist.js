import React, { useState } from "react";

import "./style.css";
import PokeListItem from "../PokeListItem/PokeListItem";

function PokeList(props) {
  const { pokelist, getData } = props;
  const [searchTerm, setSearchTerm] = useState("");

  const handleAddCart = (name, price, image, quantity) => {
    getData(name, price, image, quantity);
  };

  const handleChange = (e) => {
    setSearchTerm(e.target.value);
  };

  return (
    <>
      <input
        id="searchInput"
        type="text"
        placeholder="Pesquisar"
        value={searchTerm}
        onChange={handleChange}
      />
      <ul id="pokelist">
        {pokelist.map((pokemon, index) => {
          if (!pokemon.name.includes(searchTerm)) {
            return null;
          }

          return <PokeListItem key={index} pokemon={pokemon} addCart={handleAddCart} />;
        })}
      </ul>
    </>
  );
}

export default PokeList;
