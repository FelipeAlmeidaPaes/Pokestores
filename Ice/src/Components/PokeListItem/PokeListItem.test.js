import React from "react";
import PokeList from "./PokeListItem";
import renderer from "react-test-renderer";
import PokeListItem from "./PokeListItem";

const pokemon = {
  name: "cloyster",
  order: 131,
  sprites: {
    front_default:
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/91.png",
  },
  abilities: [
    {
      ability: {
        name: "shell-armor",
      },
    },
    {
      ability: {
        name: "overcoat",
      },
    },
  ],
};

test("snapshot", () => {
  const tree = renderer.create(<PokeListItem pokemon={pokemon} />).toJSON();
  expect(tree).toMatchSnapshot();
});
