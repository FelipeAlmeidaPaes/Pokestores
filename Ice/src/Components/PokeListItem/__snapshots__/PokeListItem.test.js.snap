// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<li
  className="poke"
>
  <img
    alt="pokemon.img"
    className="image"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/91.png"
  />
  <p
    className="name"
  >
    cloyster
  </p>
  <p
    className="price"
  >
    <b>
      R$ 
      131,00
    </b>
  </p>
  <div
    className="form-control"
  >
    <label>
      qtd:
    </label>
    <select
      className="select__quantity"
      onChange={[Function]}
    >
      <option
        value={1}
      >
        1
      </option>
      <option
        value={2}
      >
        2
      </option>
      <option
        value={3}
      >
        3
      </option>
      <option
        value={4}
      >
        4
      </option>
      <option
        value={5}
      >
        5
      </option>
      <option
        value={6}
      >
        6
      </option>
    </select>
  </div>
  <button
    className="addCartButton"
    onClick={[Function]}
  >
    Adicionar
  </button>
  <span
    className="tooltip-text"
  >
    habilidades: 
    shell-armor, overcoat
  </span>
</li>
`;
